tests/test_core.py

```diff
@@ -700,3 +701,49 @@ def test_import_image(self, openshift, tags, imagestream_name, expect_update, ex
             .with_args(post_url, data=JsonMatcher(stream_import_json), use_json=True))
 
         assert openshift.import_image(imagestream_name, stream_import, tags=tags) is expect_import
+
+    @pytest.mark.parametrize(('image_status', 'expect_retry'), (  # noqa:F811
+        ({'status': 'Failure', 'code': 500, 'reason': 'InternalError'}, True),
+        ({'status': 'Failure', 'code': 400, 'reason': 'BadRequest'}, False),
+    ))
+    def test_import_image_retry(self, openshift, image_status, expect_retry):
+        imagestream_name = TEST_IMAGESTREAM
+
+        # Load example API response
+        this_file = inspect.getfile(TestCheckResponse)
+        this_dir = os.path.dirname(this_file)
+        json_path = os.path.join(this_dir, "mock_jsons", openshift._con.version,
+                                 'imagestreamimport.json')
+        with open(json_path) as f:
+            content_json = json.load(f)
+
+        # Assume mocked data contains a good response
+        good_resp = HttpResponse(200, {}, content=json.dumps(content_json).encode('utf-8'))
+
+        # Create a bad response
+        for image in content_json['status']['images']:
```

Comment:
This assumes .status.images has at least one element, and it also sets the 'bad' status for all of those images.
Instead, how about this?:
```
# Only update first image status.
# A single server error should cause retry.
image = content_json['status']['images'][0]
```

Reply:
Good call! Changed it.